lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -259,205 +272,224 @@ def run(self, **kwargs):
+                        raise RuntimeError("Could not validate user and password")
+                    valid_connection_check = EnclosureInfo(
+                                                host_in_pillar,
+                                                user,
+                                                password,
+                                                port_in_pillar
+                                            ).connection_status()
+                    if not valid_connection_check:
+                        self.logger.error(
+                            f"Could not establish connection with"
+                            " controller with provided credentials"
+                        )
+                        raise ValueError("Invalid credentials provided")
                 self.update_pillar_and_conf('user', user)
-
-                # store password
                 self.update_pillar_and_conf('password', password)
+            else:
+                self.logger.error(f"Enclosure ID is not set\n"
+                    "Run following command to set the enclosure id:"
+                    "cortx_setup storage config --user <user>"
+                    " --password <passwd> --ip <ip> --port <port>")
+                raise RuntimeError(
+                    "Cannot set mode, ip and port without enclosure id"
+                )
 
-                # store ip and port as primary
-                self.update_pillar_and_conf('ip', ip)
-                self.update_pillar_and_conf('port', port)
+        if ip is not None or port is not None:
+            if self.mode is None:
+                self.logger.exception(
+                    f"mode is missing, please provide --mode argument"
+                )
+                raise RuntimeError("Incomplete arguments provided")
             else:
-                self.refresh_key_map()
-                # Handle following cases:
-                # 1. only user or the password is provided
-                # 2. only ip or port is provided
-                # cortx_setup storage config --user
-                # cortx_setup storage config --password
-                # cortx_setup storage config --ip
-                # cortx_setup storage config --port
-
-                if (user and not password) or (password and not user):
-                    self.logger.error(
-                        f"Please provide 'user' and 'passowrd' together")
-                    raise RuntimeError("Imcomplete arguments provided")
+                # This is already handled in 'mode' case
+                pass
+
+        if controller_type is not None:
+            valid_ctrl_type = ['gallium', 'indium']
+            if setup_type == "HW" and controller_type not in valid_ctrl_type:
+                self.logger.error(
+                "Invalid controller provided, please provide the"
+                " supported controller [gallium or indium]")
```

Comment:
Lets not hard-code it in message, the valid controller types might change later as well

Reply:
Addressed